src/ol_infrastructure/substructure/keycloak/__main__.py

```diff
@@ -334,85 +333,147 @@
+)
+oidc_attribute_importer_identity_provider_mapper = (
+    keycloak.AttributeImporterIdentityProviderMapper(
+        "map-touchstone-saml-email-attribute",
         realm=ol_apps_realm.id,
-        alias="touchstone-idp",
-        display_name="MIT Touchstone",
-        entity_id=f"{keycloak_url}/realms/olapps",
+        attribute_name="email",
+        identity_provider_alias=ol_apps_touchstone_saml_identity_provider.alias,
+        user_attribute="email",
+        extra_config={
+            "syncMode": "INHERIT",
+        },
+        opts=resource_options,
+    ),
+    keycloak.AttributeImporterIdentityProviderMapper(
+        "map-touchstone-saml-last-name-attribute",
+        realm=ol_apps_realm.id,
+        attribute_name="sn",
+        identity_provider_alias=ol_apps_touchstone_saml_identity_provider.alias,
+        user_attribute="lastName",
+        extra_config={
+            "syncMode": "INHERIT",
+        },
+        opts=resource_options,
+    ),
+    keycloak.AttributeImporterIdentityProviderMapper(
+        "map-touchstone-saml-first-name-attribute",
+        realm=ol_apps_realm.id,
+        attribute_name="givenName",
+        identity_provider_alias=ol_apps_touchstone_saml_identity_provider.alias,
+        user_attribute="firstName",
+        extra_config={
+            "syncMode": "INHERIT",
+        },
+        opts=resource_options,
+    ),
+)
+# Touchstone SAML [END]
+
+if stack_info.env_suffix in ["ci", "qa"]:
+    # OL-DEV-FAKE-TOUCHSTONE [START] # noqa: ERA001
+    ol_apps_dev_fake_touchstone_ci_identity_provider = keycloak.saml.IdentityProvider(
+        "fake-touchstone",
+        realm=ol_apps_realm.id,
+        alias="fake-touchstone",
+        display_name="Fake Touchstone",
+        entity_id="http://www.okta.com/exkcta3wbyYMdAMAP5d7",
         name_id_policy_format="Unspecified",
         force_authn=False,
         post_binding_response=True,
         post_binding_authn_request=True,
         principal_type="ATTRIBUTE",
         principal_attribute="urn:oid:1.3.6.1.4.1.5923.1.1.1.6",
-        single_sign_on_service_url="https://idp.mit.edu/idp/profile/SAML2/POST/SSO",
+        single_sign_on_service_url=keycloak_config.get_object(
+            "fake_touchstone_single_sign_on_service_url"
+        ),
         trust_email=True,
         validate_signature=True,
-        signing_certificate=mit_touchstone_cert,
+        signing_certificate=keycloak_config.get_object("fake_touchstone_sig_cert"),
         want_assertions_encrypted=True,
         want_assertions_signed=True,
         opts=resource_options,
         first_broker_login_flow_alias=ol_touchstone_first_login_flow.alias,
+        hide_on_login_page=False,
     )
     oidc_attribute_importer_identity_provider_mapper = (
         keycloak.AttributeImporterIdentityProviderMapper(
-            "map-touchstone-saml-email-attribute",
+            "map-fake-touchstone-ci-saml-email-attribute",
```

Comment:
This probably needs to be otherwise it'll throw an error for using the same name multiple times.
```suggestion
            f"map-fake-touchstone-{stack_info.env_suffix}-saml-email-attribute",
```